archiver_test.go

```diff
@@ -39,6 +39,10 @@ func symmetricTest(t *testing.T, name string, ar Archiver) {
 		t.Fatalf("making archive: didn't expect an error, but got: %v", err)
 	}
 
+	if !ar.Match(outfile) {
+		t.Fatalf("identifying format should be 'true', but got 'false'")
+	}
```

Comment:
It looks like this only checks the file extension and not the file header code, maybe we should add another few lines to the test to call those functions to verify they are working too.